projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -0,0 +1,327 @@
+
+        // Main LawsuitActivity path:
+        // app_name/lawsuit_main_dir/
+        lawsuitMainPath = Paths.get(Environment.getExternalStorageDirectory().getPath(),getString(R.string.app_name),LAWSUIT_MAIN_DIR_NAME).toString();
+        if (!dirExists(lawsuitMainPath)){
+            File directory = new File(lawsuitMainPath);
+            directory.mkdirs();
+        }
+
+        // Template's dir:
+        // app_name/lawsuit_main_dir/lawsuit_template_dir/
+        // TODO: Currently template's file is been stored in user's device storage. Check to see if this should be part of the apk package.
+        lawsuitTemplateDirPath = Paths.get(Environment.getExternalStorageDirectory().getPath(),getString(R.string.app_name),LAWSUIT_MAIN_DIR_NAME,LAWSUIT_TEMPLATE_DIR_NAME).toString();
+        if (!dirExists(lawsuitTemplateDirPath)){
+            File directory = new File(lawsuitTemplateDirPath);
+            directory.mkdirs();
+        }
+
+        // Template file's path:
+        try {
+            lawsuitTemplateFilePath = Paths.get(lawsuitTemplateDirPath, LAWSUIT_TEMPLATE_FILE_NAME).toString();
+        }
+        catch (Exception e){
+            Toast.makeText(this, "No template file found for the lawsuit",
+                    Toast.LENGTH_LONG).show();
+        }
+        // User's Lawsuit dir:
+        // app_name/lawsuit_output_dir/
+        lawsuitOutputPath = Paths.get(Environment.getExternalStorageDirectory().getPath(),getString(R.string.app_name)+" - "+ LAWSUIT_OUTPUT_DIR_NAME).toString();
+        if (!dirExists(lawsuitOutputPath)){
+            File directory = new File(lawsuitOutputPath);
+            directory.mkdirs();
+        }
+    }
+
+    /**
+     * Cehcks if a directory given in the parameter exists.
+     * @param path Directory to look for.
+     * @return True - Exists, False - Doesn't exists.
+     */
+    public boolean dirExists(String path)
+    {
+        boolean res = false;
+        File dir = new File(path);
+        if(dir.exists() && dir.isDirectory())
+            res = true;
+        return res;
+    }
+
+    // Check permissions: TODO: Check if this should be done at an earlier activity.
+    public boolean isReadStoragePermissionGranted() {
+        if (Build.VERSION.SDK_INT >= 23) {
+            if (checkSelfPermission(Manifest.permission.READ_EXTERNAL_STORAGE)
+                    == PackageManager.PERMISSION_GRANTED) {
+                Log.w(TAG, "Permission is granted1");
+                return true;
+            } else {
+
+                Log.v(TAG, "Permission is revoked1");
+                ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.READ_EXTERNAL_STORAGE}, 3);
+                return false;
+            }
+        } else { //permission is automatically granted on sdk<23 upon installation
+            Log.v(TAG, "Permission is granted1");
+            return true;
+        }
+    }
+
+    public boolean isWriteStoragePermissionGranted() {
```

Comment:
Looks like this doesn't need to be public.

Reply:
Changed this one to private. All other functions are set to protected. Should functions such as fillInTemplate() that is been used as part of createPdf() function, be set to private?

Reply:
I would set things to private, not protected. You don't plan to have this class derived from. If in the future it will happen, the dev will change private to protected where needed.